include/deal.II/base/std_cxx11/array.h

```diff
@@ -16,6 +16,7 @@
 #ifndef dealii_std_cxx11_array_h
 #define dealii_std_cxx11_array_h
 
+#warning "This file is deprecated. Simply use the corresponding C++11 header <array>."
```

Comment:
I think that for MSVC compatibility we need to use `DEAL_II_WARNING` which is in `config.h`.